### PR TITLE
fix: Exclude closing overlays from __attachedInstances

### LINKED ISSUE #176: Closing multiple dialogs and pointer-events
- Description of the bug
Whenever multiple `Dialog` instances are rendered (> 2) and the top two are closed at the same time, the underlying `Dialog` cannot be interacted with - it still has `pointer-events: none`.

- Minimal reproducible example
As an example I downloaded the `Simple App` starter, added a `View` (`extends HorizontalLayout`, the same as other `View`s in the starter)  and in it's constructor written the following code:
```java
add(new Button("Open Dialog 1", e -> {
    final Dialog dialog1 = new Dialog();
    dialog1.setCloseOnOutsideClick(false);
    dialog1.add(new Label("This is Dialog 1"));
    dialog1.add(new Button("Open Dialog 2", e1 -> {
        final Dialog dialog2 = new Dialog();
        dialog2.setCloseOnOutsideClick(false);
        dialog2.add(new Text("This is Dialog 2"));
        dialog2.add(new Button("Open Dialog 3", e2 -> {
            final Dialog dialog3 = new Dialog();
            dialog3.setCloseOnOutsideClick(false);
            dialog3.add(new Text("This is Dialog 3"));
            dialog3.add(new Button("Close Dialogs 2 and 3", e3 -> {
                dialog3.close();
                dialog2.close();
            }));
            dialog3.open();
        }));
        dialog2.open();
    }));
    dialog1.open();
}));
setSizeFull();
setJustifyContentMode(JustifyContentMode.CENTER);
setAlignItems(Alignment.CENTER);
```

The `setCloseOnOutsideClick(false)` is set so the issue is more apparent.

The first two `Dialog`s open and close fine on their own, but when clicking on the button that closes `Dialog`s 2 and 3, the first `Dialog` still has `pointer-events: none` and cannot be interacted with.

- Expected behavior
After closing multiple `Dialog`s at once the top `Dialog` should not have `pointer-events: none`.

- Actual behavior
After closing multiple `Dialog`s at once the top `Dialog` still has `pointer-events: none` and cannot be interacted with. Closing it with outside click or ESC still work as expected.

- Temporary fix
The only fix right now is to downgrade the `vaadin-overlay` package to `3.2.18`. On that version everything works fine.

- Versions:
    - Vaadin `14.1.21` / Flow version `2.1.6`
    - Java `8`, `11`
    - OS version: `macOS Catalina 10.15.3` 
    - Browsers:
        - Chrome `80.0.3987.149`
        - Firefox `74.0`
        - Safari `13.0.5`
        - Edge `80.0.361.69`

### PR DESCRIPTION
**Pull request by tomivirkki**
Fixes: #176

### CODE REVIEW

File: src/vaadin-overlay.html [hunk 1]
```diff
@@ -736,6 +736,7 @@
       static get __attachedInstances() {
         return Array.from(document.body.children)
           .filter(el => el instanceof OverlayElement)
+          .filter(el => !el.hasAttribute('closing'))
```
Comment (tomivirkki) [inline src/vaadin-overlay.html]: `__attachedInstances` is used in `_exitModalState` to restore pointer events for all overlays from the topmost until the first modal overlay is [reached](https://github.com/vaadin/vaadin-overlay/blob/master/src/vaadin-overlay.html#L818).

In the issue/test, both of the two closing modal overlays considered each other to be the frontmost modal one and thus the actual remaining overlay never got it's pointer-events restored.

The fix excludes "closing" overlays from the `__attachedInstances` so they're not considered to be "attached" while the closing animation is still active.

Comment (tomivirkki) [reply src/vaadin-overlay.html]: An alternative fix would be to invoke `_exitModalState` inside the `finishClosing` block (but I think including "closing" overlays in the array might be more error-prone and a potential source of similar issues later).

**Review by pekam — COMMENTED** [review]
I'm not so familiar with the overlay's animation logic, so just to make sure, do I understand correctly:

When there's no CSS animation set for the overlay, it gets detached from the DOM synchronously and the bug doesn't occur. When the overlay has CSS animation, it waits for `animationend` event before detaching from DOM. In this case, the bug occurs because the previously closed overlay is still in the DOM. And Lumo theme for `vaadin-dialog` sets the opening/closing animations to overlay?

**Review by pekam — APPROVED** [review]
LGTM

I verified that this fixes the `vaadin-dialog-flow` use case with the Java code provided in the ticket description.

**Comment by tomivirkki** [comment]
> When there's no CSS animation set for the overlay, it gets detached from the DOM synchronously and the bug doesn't occur. When the overlay has CSS animation, it waits for animationend event before detaching from DOM. In this case, the bug occurs because the previously closed overlay is still in the DOM. And Lumo theme for vaadin-dialog sets the opening/closing animations to overlay?

Yes, this is how it works. The bug only occurs when there's a closing animation.

File: src/vaadin-overlay.html (hunk 1 above)
Comment (web-padawan) [reply src/vaadin-overlay.html]: nit: we could use existing `.filter()` call instead of adding another one. Not a blocker.

**Review by web-padawan — APPROVED** [review]

**Pull request merged**